semcore/d3-chart/src/utils.ts

```diff
@@ -338,3 +333,20 @@ export const calculateBubbleDomain = (
 
   return [min, max];
 };
+
+interface PlotEventEmitterEmit {
+  (event: 'setTooltipVisible', visible: boolean): void;
+  (event: 'setTooltipPosition', x: number, y: number): void;
+}
+type Unsubscribe = () => void;
+interface PlotEventEmitterSubscribe {
+  (event: 'setTooltipVisible', callback: (visible: boolean) => void): Unsubscribe;
+  (event: 'setTooltipPosition', callback: (x: number, y: number) => void): Unsubscribe;
+}
+type PlotEventEmitter = {
+  new (): {
+    emit: PlotEventEmitterEmit;
+    subscribe: PlotEventEmitterSubscribe;
+  };
+};
+export const PlotEventEmitter = EventEmitter as any as PlotEventEmitter;
```

Comment:
What do you think about `EventEmitter<Emit, Subscribe>` ? And some changes in types in utils for this?